source/extensions/clusters/redis/redis_cluster.h

```diff
@@ -0,0 +1,231 @@
+#pragma once
+
+#include <array>
+#include <atomic>
+#include <chrono>
+#include <cstdint>
+#include <functional>
+#include <list>
+#include <memory>
+#include <string>
+#include <tuple>
+#include <utility>
+#include <vector>
+
+#include "envoy/api/api.h"
+#include "envoy/api/v2/cds.pb.h"
+#include "envoy/api/v2/core/base.pb.h"
+#include "envoy/api/v2/endpoint/endpoint.pb.h"
+#include "envoy/config/cluster/redis/redis_cluster.pb.h"
+#include "envoy/config/cluster/redis/redis_cluster.pb.validate.h"
+#include "envoy/config/typed_metadata.h"
+#include "envoy/event/dispatcher.h"
+#include "envoy/event/timer.h"
+#include "envoy/http/codec.h"
+#include "envoy/local_info/local_info.h"
+#include "envoy/network/dns.h"
+#include "envoy/runtime/runtime.h"
+#include "envoy/secret/secret_manager.h"
+#include "envoy/server/transport_socket_config.h"
+#include "envoy/ssl/context_manager.h"
+#include "envoy/stats/scope.h"
+#include "envoy/thread_local/thread_local.h"
+#include "envoy/upstream/cluster_manager.h"
+#include "envoy/upstream/health_checker.h"
+#include "envoy/upstream/load_balancer.h"
+#include "envoy/upstream/locality.h"
+#include "envoy/upstream/upstream.h"
+
+#include "common/common/callback_impl.h"
+#include "common/common/enum_to_int.h"
+#include "common/common/logger.h"
+#include "common/config/metadata.h"
+#include "common/config/well_known_names.h"
+#include "common/network/address_impl.h"
+#include "common/network/utility.h"
+#include "common/stats/isolated_store_impl.h"
+#include "common/upstream/cluster_factory_impl.h"
+#include "common/upstream/load_balancer_impl.h"
+#include "common/upstream/outlier_detection_impl.h"
+#include "common/upstream/resource_manager_impl.h"
+#include "common/upstream/upstream_impl.h"
+
+#include "server/transport_socket_config_impl.h"
+
+#include "extensions/clusters/well_known_names.h"
+#include "extensions/filters/network/common/redis/client.h"
+#include "extensions/filters/network/common/redis/client_impl.h"
+#include "extensions/filters/network/common/redis/codec.h"
+
+#include "absl/synchronization/mutex.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Clusters {
+namespace Redis {
+
+class RedisCluster : public Upstream::BaseDynamicClusterImpl {
```

Comment:
Can you add a medium/large doc comment on what this does at a high level? It will help future readers that might be be familiar with Redis Cluster.

Reply:
I added an explanation in the protobuf definition- I'll add a bit here too and not try to be too redundant.